compiler/rustc_hir_analysis/src/delegation.rs

```diff
@@ -0,0 +1,255 @@
+use rustc_data_structures::fx::FxHashMap;
+use rustc_hir::def::DefKind;
+use rustc_hir::def_id::{DefId, LocalDefId};
+use rustc_middle::ty::fold::{TypeFoldable, TypeFolder, TypeSuperFoldable};
+use rustc_middle::ty::{self, Ty, TyCtxt};
+use rustc_span::ErrorGuaranteed;
+use rustc_type_ir::visit::TypeVisitableExt;
+
+type RemapTable = FxHashMap<u32, u32>;
+
+struct IndicesFolder<'tcx, 'a> {
+    tcx: TyCtxt<'tcx>,
+    remap_table: &'a RemapTable,
```

Comment:
```suggestion
    remap_table: RemapTable,
```
The reference is not currently necessary.